.github/workflows/pull_request_for_xcm.yaml

```diff
@@ -0,0 +1,85 @@
+name: Check pull requests
+
+on:
+  pull_request:
+    paths:
+      - 'xcm/**'
+
+jobs:
+  create-comment-with-changes:
+    runs-on: ubuntu-latest
+    name: Create a comment with xcm changed data
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: 🛠 Set up actual paths
+        uses: ./.github/workflows/setup-path
+
+      - name: Was transfers.json changed?
+        id: prod
+        uses: tj-actions/changed-files@v18.2
+        with:
+          files: |
+            ${{ env.XCM_JSON_PATH }}
+
+      - name: Was transfers_dev.json changed?
+        id: dev
+        uses: tj-actions/changed-files@v18.2
+        with:
+          files: |
+            ${{ env.XCM_JSON_PATH }}
+
+      - name: Install dependencies
+        run: pip install -r requirements.txt
+
+      - name: Generate message for PROD
+        if: steps.prod.outputs.any_changed == 'true'
+        run: |
+          output=$(python scripts/print_xcm_changes.py prod)
+          echo "XCM_CHANGES<<EOF" >> $GITHUB_ENV
+          echo "$output" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Generate message for DEV
+        if: steps.dev.outputs.any_changed == 'true'
+        run: |
+          output=$(python scripts/print_xcm_changes.py dev)
+          echo "DEV_XCM_CHANGES<<EOF" >> $GITHUB_ENV
+          echo "$output" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
```

Comment:
I think it is better to use files to share data between steps. Envs should be ok for now with only one concern - do they have a length limit?

Reply:
Yes you are right, there is some limitation for the env size, updated solution to store changes in file